src/components/experiment-tracking/details/details.js

```diff
@@ -40,13 +44,26 @@ const Details = ({
 
   return (
     <>
-      <RunDetailsModal
-        runMetadataToEdit={runMetadataToEdit}
-        runs={runMetadata}
-        setShowRunDetailsModal={setShowRunDetailsModal}
-        theme={theme}
-        visible={showRunDetailsModal}
-      />
+      <ButtonTimeoutContextProvider>
+        {showRunDetailsModal && (
+          <RunDetailsModal
+            runMetadataToEdit={runMetadataToEdit}
+            runs={runMetadata}
+            setShowRunDetailsModal={setShowRunDetailsModal}
+            theme={theme}
+            visible={showRunDetailsModal}
+          />
+        )}
+        {showRunExportModal && (
+          <RunExportModal
+            theme={theme}
+            visible={showRunExportModal}
+            setShowRunExportModal={setShowRunExportModal}
+            runMetadata={runMetadata}
+            runTrackingData={runTrackingData}
+          />
+        )}
```

Comment:
I asked about this yesterday when you were getting a sense check on this approach, and now looking at this work locally it seems that this _does_ change the behavior of the modals animating in and out.

If you look on the `main` branch you'll see that when they enter there's a fade in and up, and when they disappear it's down and out. We'll need to keep that UX here, too, which means we'll have to find a way not to do this `showRunExportModal &&` or do that and still animate.

Reply:
hey @tynandebold such a good spot, you're right. We should not remove the animation, sorry i didnt realise it yesterday. Actually, i can just use the existing props `visible` to apply the context, rather than check with props to render a specific component 

Let me know what you think about the latest changes? Happy to discuss further